tests/logger_structlog/test_local_decorating.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import platform
+
+from newrelic.api.application import application_settings
+from newrelic.api.background_task import background_task
+from newrelic.api.time_trace import current_trace
+from newrelic.api.transaction import current_transaction
+from testing_support.fixtures import reset_core_stats_engine
+from testing_support.validators.validate_log_event_count import validate_log_event_count
+from testing_support.validators.validate_log_event_count_outside_transaction import validate_log_event_count_outside_transaction
+
+
+def set_trace_ids():
+    txn = current_transaction()
+    if txn:
+        txn._trace_id = "abcdefgh12345678"
+    trace = current_trace()
+    if trace:
+        trace.guid = "abcdefgh"
+
+def exercise_logging(logger):
+    set_trace_ids()
+
+    logger.warning("C")
+
+
+def get_metadata_string(log_message, is_txn):
```

Comment:
That's fair! We wouldn't be able to fully hardcode this into the tests since it includes variables like the host. We could construct the string each time in the inside and outside transaction tests though to skip the if-else logic. Another option is to move this into conftest as a fixture. This function is used in our other two logging test suites so they can import from this test suite and then we specify the test suite name in `Python%20Agent%20Test%20%28logger_structlog%29`